### PR TITLE
CIVDT-964: update to create case service to accommodate bulk entry

### PR DESCRIPTION
**Pull request by tvaidotas**

### CODE REVIEW

File: test/services/submissionService/BuildEntryDetailsSpec.scala [hunk 1]
```diff
@@ -30,11 +30,19 @@ class BuildEntryDetailsSpec extends SpecBase with MockIvdSubmissionConnector wit
   "buildEntryDetails" when {
 
     "called with valid User Answers" should {
+
       "return expect json" in {
```
Comment (macgrewal) [inline test/services/submissionService/BuildEntryDetailsSpec.scala:34]:
```suggestion
      "return expect json for single entry" in {
```

File: test/services/submissionService/BuildReasonsDetailsSpec.scala [hunk 2]
```diff
@@ -29,11 +29,19 @@ class BuildReasonsDetailsSpec extends SpecBase with MockIvdSubmissionConnector w
   "buildReasonsDetails" when {
 
     "called with valid User Answers" should {
+
       "return expect json" in {
```
Comment (macgrewal) [inline test/services/submissionService/BuildReasonsDetailsSpec.scala:33]:
```suggestion
      "return expect json for a single entry disclosure" in {
```

File: test/services/submissionService/BuildReasonsDetailsSpec.scala [hunk 3]
```diff
@@ -29,11 +29,19 @@ class BuildReasonsDetailsSpec extends SpecBase with MockIvdSubmissionConnector w
...
         lazy val result = service.buildReasonsDetails(importerSubmission)
 
         result mustBe Json.parse(reasonsDetailsJson)
       }
+
+      "return expect json in bulk submission" in {
```
Comment (macgrewal) [inline test/services/submissionService/BuildReasonsDetailsSpec.scala:39]:
```suggestion
      "return expect json for a bulk entry disclosure" in {
```

**Review by macgrewal — CHANGES_REQUESTED** [review]
Minor code refactoring suggestions and updating test descriptions but other than that looks good.

File: app/services/SubmissionService.scala [hunk 4]
```diff
@@ -83,18 +83,27 @@ class SubmissionService @Inject()(ivdSubmissionConnector: IvdSubmissionConnector
...
+      )
+    } else {
+      Json.obj(
+        "userType" -> data.userType,
+        "isBulkEntry" -> isBulkEntry,
+        "isEuropeanUnionDuty" -> data.acceptedBeforeBrexit,
+        "entryDetails" -> data.entryDetails,
+        "customsProcessingCode" -> customsProcessingCode
+      )
+    }
```
[the remark this answers is not in the document]
Comment (tvaidotas) [reply app/services/SubmissionService.scala:106]: We had it this way but decided that the current way was better, so not sure how to address this...

**Review by macgrewal — APPROVED** [review]

**Pull request merged**